synapse/config/tls.py

```diff
@@ -110,8 +111,15 @@ def read_config(self, config, config_dir_path, **kwargs):
         # Support globs (*) in whitelist values
         self.federation_certificate_verification_whitelist = []
         for entry in fed_whitelist_entries:
+
+            # Don't try and IDNA encode globs
+            if entry[0:2] == "*.":
```

Comment:
So this changes it so we only support globs as the first part? I think I'd rather make people put encoded domain names in their config than break globs

Reply:
I'm not sure that non-subdomain globs are actually functionally useful?